components/skus/browser/skus_sdk_context_impl.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_SKUS_BROWSER_SKUS_SDK_CONTEXT_IMPL_H_
+#define BRAVE_COMPONENTS_SKUS_BROWSER_SKUS_SDK_CONTEXT_IMPL_H_
+
+#include <memory>
+#include <string>
+
+#include "base/memory/scoped_refptr.h"
+#include "brave/components/skus/browser/rs/cxx/src/shim.h"
+
+class PrefService;
+
+namespace network {
+class SharedURLLoaderFactory;
+}  // namespace network
+
+namespace user_prefs {
+class PrefRegistrySyncable;
+}  // namespace user_prefs
+
+namespace skus {
+class SkusSdkFetcher;
+}  // namespace skus
+
+namespace skus {
+
+// Context object used with the SKU SDK to provide 1) key/value pair storage
+// and 2) the fetcher used for contacting the SKU SDK endpoint via HTTPS.
+//
+// In the .cc, there are implementations for global methods originally defined
+// in `rs/cxx/src/shim.h`. These implementations are called from
+// Rust and will pass this context object along, so that the results can be
+// persisted.
+class SkusSdkContextImpl : public SkusSdkContext {
+ public:
+  // TODO(bridiver) - this is not the right place for this
+  static void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry);
```

Comment:
this should move to the service or override `BrowserContextKeyedServiceFactory::RegisterProfilePrefs` in the factory

Reply:
DONE - moved `RegisterProfilePrefs` into factory and then added `skus::SkusSdkServiceFactory::GetInstance()` to `EnsureBrowserContextKeyedServiceFactoriesBuilt`
See https://github.com/brave/brave-core/pull/10847/commits/780676bbebb2283387763229a4592c4eb4eadb9d